src/main/java/betterquesting/handlers/PlayerContainerListener.java

```diff
@@ -69,7 +130,30 @@ public void sendWindowProperty(@Nonnull Container container, int i, int i1) {
     public void sendAllWindowProperties(@Nonnull Container container, @Nonnull IInventory iInventory) {
     }
 
-    private void updateTasks() {
-        EventHandler.schedulePlayerInventoryCheck(player);
+    /**
+     * Schedules checking this player's inventory after the configured delay.
+     * Deduplicates requests to avoid scanning it multiple times per tick.
+     */
+    private void scheduleChange() {
+        // Sanity check to ensure this only happens on the main server thread
+        if (!FMLCommonHandler.instance().getMinecraftServerInstance().isCallingFromMinecraftThread()) {
+            FMLCommonHandler.instance().getMinecraftServerInstance().addScheduledTask(this::scheduleChange);
```

Comment:
the way the comment it worded implies that this if statement should `return` and prevent the rest of this method from running